test/Microsoft.AspNetCore.Mvc.Core.Test/Builder/MvcApplicationBuilderExtensionsTest.cs

```diff
@@ -49,5 +56,61 @@ public void UseMvcWithEndpoint_ThrowsInvalidOperationException_IfMvcMarkerServic
                 "in the application startup code.",
                 exception.Message);
         }
+
+        [Fact]
+        public void UseMvc_GlobalRoutingDisabled_NoEndpointInfos()
+        {
+            // Arrange
+            var services = new ServiceCollection();
+            services.AddSingleton<DiagnosticSource>(new DiagnosticListener("Microsoft.AspNetCore"));
```

Comment:
Is this expected? I found `UseMvc` with `AddMvcCore` would fail without registering a `DiagnosticSource`.

Reply:
Yes, there's a small set of services we expect Hosting to register for us.